<a href="https://colab.research.google.com/github/tayyba6/ML_Internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My lane: Refresh / Content Opportunity Scoring

I chose this lane because I want to prioritize pages for review so that limited content-team time can be focused on the pages most likely to need attention. The starter data provides signals that can be used to build and evaluate a practical ranking approach.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Question:** Which pages should be prioritized for review first for possible content improvement?

**Decision:** Decide which pages should receive attention first when review time is limited.

**Action:** Review the highest-priority pages and decide whether each page needs a refresh, protection, monitoring, or no action.

**Cost of a wrong decision:** A page that is incorrectly prioritized could waste review time, while a page that is incorrectly deprioritized could cause a potentially important content opportunity to be missed.

In [ ]:
import os
import sys
import subprocess

REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if not os.path.isdir(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)

os.chdir(REPO_DIR)

print("Working directory:", os.getcwd())

Working directory: /content/flyrank-ml-internship-starter


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Pages:", len(df))
print("Columns:", len(df.columns))

declining_rate = (df["trend_direction"] == "down").mean()
print("Declining pages:", round(declining_rate, 3))

Pages: 30000
Columns: 44
Declining pages: 0.542


The starter dataset contains 30,000 pages and 44 columns. About 54.2% of the pages have a downward trend direction, which provides a useful signal to investigate when building a ranking for review prioritization. These numbers make the content-review prioritization problem worth exploring further.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This work can identify observed patterns in the starter data and produce a directional ranking that can help prioritize pages for human review. The ranking is decision-support, not an automatic decision about what a page needs. It does not prove that any factor causes a page to decline, that a page definitely needs a refresh, or that changing a page will improve its performance.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.